paimon-spark/paimon-spark-ut/src/test/java/org/apache/paimon/spark/catalog/functions/BucketFunctionTest.java

```diff
@@ -0,0 +1,318 @@
+        spark =
+                SparkSession.builder()
+                        .master("local[2]")
+                        .config(
+                                "spark.sql.extensions",
+                                PaimonSparkSessionExtensions.class.getName())
+                        .getOrCreate();
+        spark.conf().set("spark.sql.catalog.paimon", SparkCatalog.class.getName());
+        spark.conf().set("spark.sql.catalog.paimon.warehouse", warehousePath.toString());
+        spark.sql("CREATE DATABASE paimon.db");
+        spark.sql("USE paimon.db");
+    }
+
+    @AfterEach
+    public void after() {
+        spark.sql(String.format("DROP TABLE IF EXISTS %s", TABLE_NAME));
+    }
+
+    public static void setupTable(String... bucketColumns) {
+        String commitUser = UUID.randomUUID().toString();
+        try {
+            TableSchema tableSchema =
+                    SchemaUtils.forceCommit(
+                            new SchemaManager(LocalFileIO.create(), tablePath),
+                            new Schema(
+                                    ROW_TYPE.getFields(),
+                                    Collections.emptyList(),
+                                    Collections.emptyList(),
+                                    ImmutableMap.of(
+                                            CoreOptions.BUCKET_KEY.key(),
+                                            String.join(",", bucketColumns),
+                                            CoreOptions.BUCKET.key(),
+                                            String.valueOf(NUM_BUCKETS)),
+                                    ""));
+            FileStoreTable storeTable =
+                    FileStoreTableFactory.create(LocalFileIO.create(), tablePath, tableSchema);
+            TableWriteImpl<?> write =
+                    storeTable
+                            .newWrite(commitUser)
+                            .withIOManager(new IOManagerImpl(ioManagerPath.toString()));
+            TableCommitImpl tableCommit = storeTable.newCommit(commitUser);
+
+            write.write(randomPaimonInternalRow());
+            write.write(NULL_PAIMON_ROW);
+            tableCommit.commit(0, write.prepareCommit(false, 0));
+
+            write.close();
+            tableCommit.close();
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Test
+    public void test() {
+        validateSparkBucketFunction(INTEGER_COL);
+    }
+
+    private static void validateSparkBucketFunction(String... bucketColumns) {
+        setupTable(bucketColumns);
+        spark.sql(
+                        String.format(
+                                "SELECT id_col, __paimon_bucket as expected_bucket, paimon.bucket(%s, %s) FROM %s",
+                                NUM_BUCKETS, String.join(",", bucketColumns), TABLE_NAME))
+                .collectAsList()
+                .forEach(row -> Assertions.assertThat(row.getInt(2)).isEqualTo(row.get(1)));
```

Comment:
Added this test to compare the bucket values calculated by directly writing to the Paimon table with those calculated using the bucket function registered in the Spark function catalog, so this proves that the values calculated by the bucket function in the function catalog are correct.